File: csharp/DesignPatter.Testing.PageObjectPattern/PageObject/Pages/EliteMoviePage.cs
//-----------------------------------------------------------------------
// <copyright file="EliteMoviePage.cs" company="ShareKnowledge">
//     Copyright (c) ShareKnowledge. All rights reserved.
// </copyright>
// <author>Alejandro Perdomo</author>
//-----------------------------------------------------------------------

namespace DesignPatter.Testing.PageObjectPattern.PageObject.Pages
{
  #region Imports

  using OpenQA.Selenium;

  #endregion Imports

  internal class EliteMoviePage
  {
    #region Attributes

    private readonly IWebDriver driver;

    #endregion Attributes

    #region Constructors

    public EliteMoviePage(IWebDriver driver)
    {
      this.driver = driver;
    }

    #endregion Constructors

    #region Properties

    private IWebElement SearchFilm =>
        this.driver.FindElement(By.CssSelector(".searchfield"));

    private IWebElement FirstFilm =>
        this.driver.FindElement(By.CssSelector("a.ng-scope:nth-child(1) > img:nth-child(1)"));

    #endregion Properties

    #region Methods

    public SelectionSchedulePage SelectFilm(string film)
    {
      this.SearchFilm.SendKeys(film);
      this.FirstFilm.Click();
      return new SelectionSchedulePage(this.driver);
    }

    #endregion Methods 
  }
}

File: java/elitemovie-ui-test/src/test/java/page/EliteMoviePage.java
package page;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;

public class EliteMoviePage {
	
	private WebDriver driver;
	
	public EliteMoviePage(WebDriver driver) {
		this.driver = driver;
	}

	private WebElement getSearhFilm() {
		return this.driver.findElement(By.cssSelector(".searchfield"));
	}
	
	private WebElement getFilm() {
		return driver.findElement(By.cssSelector("a.ng-scope:nth-child(1) > img:nth-child(1)"));
	}
	
	public void selectFilm(String filmName) {
	    this.getSearhFilm().sendKeys(filmName);
	    this.getFilm().click();
	}
}
